transform/mattermost-analytics/models/intermediate/product/licenses/int_known_licenses.sql

```diff
@@ -1,26 +1,102 @@
--- List of all self-hosted license data from CWS and legacy licenses.
--- Performs deduplication in case a license exists both in CWS and legacy licenses.
+-- List of all self-hosted license data from CWS, Salesforce, telemetry and legacy licenses.
+-- Performs deduplication in case a license exists in multiple sources.
 -- In case both CWS and legacy data are found, CWS data are preferred.
 with deduped_legacy_licenses as (
+
     select
         {{ dbt_utils.star(ref('stg_licenses__licenses')) }}
     from
         {{ ref('stg_licenses__licenses') }}
     group by all
+
+), telemetry_licenses as (
+
+   select
+        license_id
+        , min(license_name) as license_name
+        , min(licensed_seats) as licensed_seats
+        , min(issued_at) as issued_at
+        , min(starts_at) as starts_at
+        , min(expire_at) as expire_at
+    from
+        {{ ref('int_server_license_daily') }}
+    group by license_id
+    -- Filter out licenses with ambiguous data
+    having
+        count(distinct licensed_seats) = 1
+        and count(distinct starts_at) = 1
+        and count(distinct expire_at) = 1
+        and count(distinct customer_id) = 1
+
+) , salesforce_licenses as (
+
+    select
+        o.opportunity_id,
+        a.account_id,
+        a.name as account_name,
+        license_key__c as license_id,
+        license_start_date__c as starts_at,
+        license_end_date__c as expire_at,
+        seats_from_name
+    from
+        {{ ref('stg_salesforce__opportunity') }} o
+        left join {{ ref('stg_salesforce__account') }} a  on o.account_id = a.account_id
+    where
+        license_key__c is not null
+        and o.stage_name = '6. Closed Won'
+        -- License reported in multiple accounts or license has invalid value - temporarily remove
+        and o.opportunity_id not in (
+            '0063p00000yaF7lAAE',
```

Comment:
might help document what these Ids represent.

Reply:
It's in the comment above. Opportunities belonging to different accounts but sharing the same license id.